### DISCUSSION TITLE
Rewind Shake

### DISCUSSION THREAD

**Discussion by wtmsuperman** [KyryloKuzyk/PrimeTween#64] [FAQ] [answered]
Shake don't have start value,how can I rewind a shake safely?
I have a common function to rewind a tween,may be not a shake,currently,I do this way:

```
if (tween.isAlive)
{
    tween.progress = 1.0f; //complete shake and do not trigger onComplete
}
else
{
    tween= CreateTween();
}
tween.progress = 0;
tween.Stop();
```

Is this the right way?

And I have some sequences need to play and rewind,Could I use `progress` like tween?

**Comment by KyryloKuzyk** [comment, marked as answer]
Hey, sorry for the late reply.

Yes, setting progress = 0f, then calling Stop() will rewind any tween to the initial value without triggering OnComplete() callback. And yes, you can do the same thing for Sequence.
```csharp
public static void RewindAndStop(Tween tween) {
    tween.progress = 0f;
    tween.Stop();
}
```
